<launch>
    <arg name="speed" default="0"/>
    <arg name="steering_speed" default="0"/>
    
    <arg name="kp" default="10" />
    <arg name="ki" default="0" />
    <arg name="kd" default="1.5" />
    
    <group ns="raceon">
        <param name="topic_name_camera_image" value="camera/image" />
        <param name="topic_name_camera_info" value="camera/camera_info" />
        <param name="topic_name_position_error" value="position/error" />
        <param name="topic_name_position_track" value="position/track" />
        <param name="topic_name_position_state" value="position/state" />
        <param name="topic_name_control" value="control" />

        <node pkg="raceon" name="camera" type="camera.py" output="screen">
            <param name="~publish_raw" value="false" />
            <param name="~resolution/width" value="640" />
            <param name="~resolution/height" value="480" />
            <param name="~fps" value="30" />
        </node>

        <node pkg="raceon" name="pos_estimation" type="pos_estimation.py" output="log">
            <param name="~scan_line_d" value="160" />
            <param name="~scan_line_u" value="40" />
            <param name="~peak_threshold_d" value="170" />
            <param name="~peak_threshold_u" value="170" />
            <param name="~track_width" value="500" />
            <param name="~camera_center" value="320" />
            <param name="~breaker" value="0" />
        </node>

        <node pkg="raceon" name="control" type="control.py" output="log">
            <param name="~motor_speed" value="$(arg speed)" />
            <param name="~steering_speed" value="$(arg steering_speed)" />
            <param name="~target" value="0" />
            <param name="~kp" value="$(arg kp)" />
            <param name="~ki" value="$(arg ki)" />
            <param name="~kd" value="$(arg kd)" />
        </node>

        <node pkg="raceon" name="actuation" type="actuation.py" output="log" />
    </group>
</launch>
